appserver/common/annotation-framework/src/main/java/org/glassfish/apf/impl/ComponentDefinition.java

```diff
@@ -142,11 +142,7 @@ static boolean isExcludedFromAnnotationProcessing(Class<?> clazz) {
         if (clazz.getPackage().getName().startsWith("java.lang")) {
             return true;
         }
-        if (clazz.getCanonicalName() == null) {
```

Comment:
Maybe it's just nanos or so, but I believe that the null check is faster than contains(null)...?

Reply:
`null`-check was introduced for other than performance reasons, and I'd like to get rid of it.

I like the pre-#24150 version better - so I'd like to restore it.
But if I do not share this liking with anyone, perhaps #24213 could be considered separately?

Reply:
Ok, ok, it's just my first idea when I asked myself why are you doing that :-)
Tests are important and they explain reasons well, it's ok to have both here.